src/extensions/behavior/SharedState/SharedState.ts

```diff
@@ -0,0 +1,14 @@
+import type {Extension} from '#core';
+import {Plugin} from '#pm/state';
+
+import {pluginSpec} from './plugin';
+
+export {SharedStateKey, type SharedStateKeySpec} from './utils';
+
+/**
+ * This extension enables a plugin that stores internal state and handles its changes via meta-information in transactions.
+ * You can get, change and subscribe to state changes using a SharedStateKey instance.
+ */
+export const SharedState: Extension = (builder) => {
```

Comment:
I suggest adding TSDoc and brief information to the documentation about what can be used for any editor extensions

Perhaps it is worth adding a mention of the similarity of the idea with codemirror facets